### DISCUSSION TITLE
Attempting to use a ET69C02 Ethernet to RS485 adaptor

### DISCUSSION THREAD

**Discussion by nigelspowage** [nathanmarlor/foxess_modbus#388] [Q&A]
Hi, I picked up a cheapo ethernet to RS485 adaptor and I'm giving it a go to use with my Foxess H1-3.7 inverter. It's not one that's currently listed as working but as it's 'what I have' I thought I'd give it a go.

It's one of these:
https://www.aliexpress.com/item/1005004766953714.html

I've configured it to appear on my network and I can get to the admin pages okay. I *think* I've set everything correctly but hopefully folks here can let me know if I've got anything wrong.

I get as far as running the modpoll command from my PC it seems to query the adaptor but I just get the 'Reply timed out' error. I've attached screenshots of the settings, I would be grateful of any suggestions on how to get this talking with the inverter.

![advanced](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/5bf20932-829c-4ac9-91dd-9ad5df92afc4)
![expand](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/c77cdb7e-fbb2-4101-8638-689e83259aed)
![lan config](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/820c597d-c697-45f0-a088-3395c66eca4d)
![main config](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/28707dea-8a5c-47e3-b39c-46d7db916aae)
![modpoll test](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/c610467f-5f3f-4c8c-abf5-67449c25f48a)
![serial config](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/cb49c8bf-8001-43e2-afbe-8e820a13cb30)
![system](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/7e7fd85a-ec5c-42c8-b84c-a28cc1f6aba3)

**Comment by canton7** [comment]
You'll need `-m enc` with modpoll, since this is just tcp to RS485, not ModbusTCP

**Reply by nigelspowage** [reply]
Thanks for the response.

I've just tried it with no change in results :(

![enc](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/e3e96097-818d-44d4-a9d8-13db131ecf51)

**Reply by canton7** [reply]
And just to make sure, you've connected to the RS485 header here, the right way around?

![Screenshot_20230805-174632](https://github.com/nathanmarlor/foxess_modbus/assets/568104/4445a602-1acf-4dc2-ac70-64c42691d5fd)

Presumably those jumpers in the middle need to be set correctly, as well?

**Reply by nigelspowage** [reply]
While the software screenshots say GT1001, it's not the board you've provided a picture of.
The board is marked ET69C02 and looks like this. There's no jumpers to change.

I've tried the wires both ways round and there's no difference to the output.

![image](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/74d3fed5-82a2-4588-a75a-4beed8543fc5)

**Reply by canton7** [reply]
Try setting the Mode to ModbusSlave on the LAN tab?

**Reply by nigelspowage** [reply]
Thanks for the suggestion, I've just tried that but unfortunately there's no difference. I did not know if I needed `-m enc` or `-m tcp` so I tried both.

![image](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/973d0097-0649-4fa6-a0d8-d2c438892344)
![image](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/c3d1b1e3-c48e-47fa-beba-db38c2481c3b)

**Reply by canton7** [reply]
From the "Status" section, it looks like it's at least trying to send stuff on over RS485. Might be worth double-checking the wiring?

**Reply by nigelspowage** [reply]
I've just unplugged the connector and tested continuity between pins 3/4 to the A+/B- on the adaptor and they're both good.


When I was wiring them in a few days ago I took this photo for reference. It is pins 3/4 for a H1-3.7-E isn't it ?
![image](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/95cb72af-8a6b-49b7-81d2-ba0d1cbe3b10)

**Reply by FozzieUK** [reply]
Yes that’s right pin 3 is 485B, pin 4 is 485A

Just a thought, have you tried it with the mode set as Modbus Host the description isn’t very clear as to what is acting as host or slave.

**Reply by nigelspowage** [reply]
Unfortunately it seems when the interface is set as Modbus Master I get the following error.
![image](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/a9a3d466-1ad1-4640-8060-ae4582dcc049)
I also increased the timeout to five seconds in case this was just slow at responding but that made no difference.
I'm starting to suspect this interface is faulty or simply not compatible.

**Reply by FozzieUK** [reply]
It’s certainly putting up a fight!, before you can it, could you just give one last try set as modbus master but change the -m tcp to -m enc

**Reply by nigelspowage** [reply]
I've just given it a go and got the same error.
![image](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/f8200c91-5a53-4e7c-b956-8dfa086e02e4)
I think when it's set as master it's listening on the RS485 side so there's nothing on the ethernet side for modpol to communicate with.

**Reply by nigelspowage** [reply]
I'm clutching at straws a little here but could it be anything to do with the cable I've used? I'm using two core low power cable because I had it lying around and it seemed a sensible size.
![image](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/51f84b1d-62d2-43d0-b290-6274ac2b8c35)
I've read in places on the wiki that 2 core cable is okay but once the length gets more than a couple of meters having twisted pairs is preferred. I don't think I'm quite at 2m but I'm probably not far off.
![image](https://github.com/nathanmarlor/foxess_modbus/assets/34656171/ea89fcc7-2a50-4c53-b342-98c4fd6aef64)

**Reply by FozzieUK** [reply]
2 core cable isn’t advisable but RS485 can go over 1000 meters when terminated correctly -  2 metres really isn’t going to be a problem..

Hmmm.. it sounds like it’s time to invest in one of the tried and tested ones in the wiki page - my preference is the Waveshare but there are plenty of cheaper ones out there.
